### PR TITLE
Start extracting sql-migration-connector SQL rendering

### PR DESCRIPTION
**Pull request by tomhoule**
A few reasons:

- Compile times
- Testability
- Potentially useful elsewhere (modularity)

### CODE REVIEW

**Review by pimeys — COMMENTED** [review]
<3

**Comment by tomhoule** [comment]
This was discussed in the engines meeting.

File: libs/sql-ddl/src/mysql.rs [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+        let mut constrained_columns = self.constrained_columns.iter().peekable();
+
+        while let Some(column) = constrained_columns.next() {
+            write!(f, "`{}`", column)?;
+
+            if constrained_columns.peek().is_some() {
+                f.write_str(", ")?;
+            }
+        }
+
```
Comment (do4gr) [inline libs/sql-ddl/src/mysql.rs:85]: Is there a reason why you're not using your join iterator here?

Comment (tomhoule) [reply libs/sql-ddl/src/mysql.rs:88]: Did just that, good suggestion, thanks!

**Pull request merged**